test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionProtocolTests.cs

```diff
@@ -19,18 +19,17 @@ public class HubConnectionProtocolTests
         public async Task SendAsyncSendsANonBlockingInvocationMessage()
         {
             var connection = new TestConnection();
-            var hubConnection = new HubConnection(connection, new JsonHubProtocol(), new LoggerFactory());
+            var hubConnection = new HubConnection(() => connection, new JsonHubProtocol(), new LoggerFactory());
```

Comment:
Create a helper method for creating HubConnection and reuse through tests. Helpers avoid updating dozens of tests each time the ctor changes.